### PR TITLE
Fixing a bunch of codacy warnings

### PR DESCRIPTION
**Pull request by hkaiser**

### CODE REVIEW

File: examples/quickstart/factorial.cpp [hunk 1]
```diff
@@ -26,8 +26,8 @@ HPX_PLAIN_ACTION(factorial, factorial_action);
 ///////////////////////////////////////////////////////////////////////////////
 std::uint64_t factorial(std::uint64_t n)
 {
-    if (0 >= n)
-        return 1;
+    if (n < 2)
+        return n;
```
Comment (msimberg) [inline examples/quickstart/factorial.cpp:30]: This isn't right if `n = 0`. It should still always return `1`.

File: examples/quickstart/hello_world_distributed.cpp [hunk 2]
```diff
@@ -47,7 +47,7 @@ std::size_t hello_world_worker(std::size_t desired)
         char const* msg = "hello world from OS-thread {1} on locality {2}\n";
 
         hpx::util::format_to(hpx::cout, msg, desired, hpx::get_locality_id())
-            << hpx::flush;
+            << std::flush;
```
Comment (msimberg) [inline examples/quickstart/hello_world_distributed.cpp:50]: Is this something we should enforce more generally (or maybe this is the last use of `hpx::flush`? I doubt it...)? And perhaps mark `hpx::flush` as deprecated?

File: libs/runtime_local/src/runtime_local.cpp [hunk 3]
```diff
@@ -952,9 +952,8 @@ namespace hpx {
             return false;
         }
 
-        bool numa_sensitive = false;
         if (std::size_t(-1) != get_worker_thread_num())
-            return numa_sensitive;
+            return false;
```
Comment (msimberg) [inline libs/runtime_local/src/runtime_local.cpp:956]: Hmm, does this function ever return anything but `false`? Either this function should query the default/current scheduler for the value or the function shouldn't exist at all. Does not need to be changed in this PR.

File: examples/quickstart/factorial.cpp (hunk 1 above)
Comment (hkaiser) [reply examples/quickstart/factorial.cpp:30]: This should never be called with `n == 0`, I think. In any case, the old test was wrong, definitely.

**Comment by hkaiser** [comment]
> Couple of questions: I've received more emails from codacy lately. Did something change in the codacy setup? 

I had to reinstall the codacy hook as they have changed things on their end. That could have caused the changes.

> Do you know if there's an option to check pull requests from forked repos as well? I think some of these have snuck in from my changes.

I don't know that, sorry.

File: examples/quickstart/hello_world_distributed.cpp (hunk 2 above)
Comment (hkaiser) [reply examples/quickstart/hello_world_distributed.cpp:50]: Yes, I think we should deprecate `hpx::flush` and all the other manipulators we have implemented. They are not needed anymore (and have not been for a long time).

File: libs/runtime_local/src/runtime_local.cpp (hunk 3 above)
Comment (hkaiser) [reply libs/runtime_local/src/runtime_local.cpp:956]: I didn't look at the semantics here. Just removed the variable that was not needed anymore.

**Comment by hkaiser** [comment]
> > Do you know if there's an option to check pull requests from forked repos as well? I think some of these have snuck in from my changes.
> 
> I don't know that, sorry.

Update: you might be able to separately integrate your repository with codacy.

File: examples/quickstart/factorial.cpp (hunk 1 above)
Comment (hkaiser) [reply examples/quickstart/factorial.cpp:30]: I also think we do the same for all the other fibonacci examples.

**Comment by msimberg** [comment]
> > > Do you know if there's an option to check pull requests from forked repos as well? I think some of these have snuck in from my changes.
> > 
> > 
> > I don't know that, sorry.
> 
> Update: you might be able to separately integrate your repository with codacy.

No worries. Yes, I can definitely do that. That'll at least work for my PRs.

File: examples/quickstart/factorial.cpp (hunk 1 above)
Comment (msimberg) [reply examples/quickstart/factorial.cpp:30]: `0! = 1` is perfectly valid. Not a very interesting input for a parallel `factorial` but it's definitely valid input. The zeroth fibonacci number is actually `0` so it makes sense there.

File: examples/quickstart/hello_world_distributed.cpp (hunk 2 above)
Comment (msimberg) [reply examples/quickstart/hello_world_distributed.cpp:50]: https://github.com/STEllAR-GROUP/hpx/issues/4736

File: libs/runtime_local/src/runtime_local.cpp (hunk 3 above)
Comment (msimberg) [reply libs/runtime_local/src/runtime_local.cpp:956]: No worries, let's not mix it into this PR.

File: examples/quickstart/factorial.cpp (hunk 1 above)
Comment (hkaiser) [reply examples/quickstart/factorial.cpp:30]: > The zeroth fibonacci number is actually 0 so it makes sense there.

Ok, then `if (n < 2) return n;` is actually correct?

Comment (msimberg) [reply examples/quickstart/factorial.cpp:30]: > Ok, then `if (n < 2) return n;` is actually correct?

Well, yes, for fibonacci. This is factorial ;)

Comment (hkaiser) [reply examples/quickstart/factorial.cpp:30]: Uhh, right. Not sure what I was thinking...

**Comment by hkaiser** [comment]
@msimberg I think this is good to go now.

**Pull request merged**